broker/actions.py

```diff
@@ -7,287 +7,285 @@
 
 logger = logging.getLogger('broker.actions')
 r = redis.Redis(config.get('redis', 'host'))
 
-TYPE_LUT = message.TYPE_LUT
-
-class MQTTSNActions:
-    @staticmethod
-    def handle(message, addr):
-        """ Perform action based on message type """
-        if message.message_type == TYPE_LUT['CONNECT']:
-            MQTTSNActions.handle_connect(message, addr)
-        elif message.message_type == TYPE_LUT['REGISTER']:
-            MQTTSNActions.handle_register(message, addr)
-        elif message.message_type == TYPE_LUT['PINGREQ']:
-            MQTTSNActions.handle_pingreq(message, addr)
-        elif message.message_type == TYPE_LUT['DISCONNECT']:
-            MQTTSNActions.handle_disconnect(message, addr)
-        elif message.message_type == TYPE_LUT['PUBLISH']:
-            MQTTSNActions.handle_publish(message, addr)
-        elif message.message_type == TYPE_LUT['SUBSCRIBE']:
-            MQTTSNActions.handle_subscribe(message, addr)
-
-    @staticmethod
-    def queue(destination, payload):
-        if payload is None:
-            logger.info('Cowardly refusing to queue empty payload')
-            return
-        
-        # add to the outgoing queue for that broker
-        logger.debug('Queued message for %s' % destination)
-        keeptime = config.getint('redis', 'keepalive') + 1
-        r.rpush('%s:queue' % (destination,),  payload)
-        r.expire('%s:queue' % (destination,), int(time.time() + keeptime))
-
-    @staticmethod
-    def create_connack(message):
-        result = bytes([3, TYPE_LUT['CONNACK'], 0]) # 0 means accepted
-        return result
-
-    @staticmethod
-    def create_regack(message):
-        topic_id = r.zrank('topics', message.topic_name)
-        topic_id_bytes = struct.pack(">H", topic_id)
-
-        result = bytes(
-                [7, TYPE_LUT['REGACK'], # 7 byte response
-                topic_id_bytes[0], topic_id_bytes[1], # the topic id
-                message.message_id[1], message.message_id[2], # the message id
-                0])
-        return result
-
-    @staticmethod
-    def create_pingresp(message):
-        result = bytes([2, TYPE_LUT['PINGRESP']])
-        return result
-
-    @staticmethod
-    def create_disconnect(message):
-        result = bytes([2, TYPE_LUT['DISCONNECT']])
-        return result
-
-    @staticmethod
-    def create_puback(message):
-        result = bytes(
-                [7, TYPE_LUT['PUBACK'], # 7 byte response
-                message.topic_id[1], message.topic_id[2], # the topic id
-                message.message_id[1], message.message_id[2], # the message id
-                0])
-        return result
-
-    @staticmethod
-    def create_suback(message, res):
-        if res is None:
-            # no match topic, tell the client it failed
-            result = 2 #congestion?
-            topic_id = (0, 0,)
+def handle(message, addr):
+    """ Perform action based on message type """
+    mty = message.message_type
+    if mty in _DISPATCH:
```

Comment:
(Understanding that this was not originally handled well, the cleanup makes it apparently missing)